File: Module 4/FactorialCalculator.cpp
/*****
* Author: Glowstudent
* Program: Factorial Calculator
*****/

#include <iostream>

using namespace std;

int main()
{
	int input;
	int max_input = 16;

	cout << "Enter any number up to 16: ";
	cin >> input;

	while (input > max_input)
	{
		cout << "Number is too large. Pick another number: ";
		cin >> input;
	}

	int i = 0, count;
	while (i < input) {
		if (i == 0) count = ++i;
		else count *= ++i;
	}

	cout << "The factorial is: " << count << endl;

	return 0;
}

File: Module 4/FactorialCalculator.java
/*****
* Author: Glowstudent
* Program: Factorial Calculator
*****/

import java.util.Scanner;

public class FactorialCalculator {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int input;
	    int max_input = 16;
        int factorial = 1;

        System.out.print("Enter any number up to 16: ");
        input = sc.nextInt();

        while (input > max_input) {
            System.out.print("Number is too large. Pick another number: ");
            input = sc.nextInt();
        }

        for (int i = 1; i <= input; i++) {
            factorial *= i;
        }

        System.out.printf("The factorial of %s is %s", input, factorial);
    }
}